web/src/components/LanguageList.tsx

```diff
@@ -0,0 +1,160 @@
+import Box from '@mui/material/Box'
+import List from '@mui/material/List'
+import TextField from '@mui/material/TextField'
+import { styled } from '@mui/material/styles'
+import React, { ReactElement, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { normalizeString } from 'shared'
+import { config } from 'translations'
+
+import useDimensions from '../hooks/useDimensions'
+import LanguageListItem from './LanguageListItem'
+import SearchInput from './SearchInput'
+
+const MobileContainer = styled('div')`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  width: 100%;
+  gap: ${props => props.theme.spacing(2)};
+`
+
+const StyledAutocomplete = styled(Autocomplete)`
+  ${props => props.theme.breakpoints.up('md')} {
+    width: 200px;
+    position: absolute;
+    top: 72px;
+    z-index: 10;
+    background-color: ${props => props.theme.palette.background.paper};
+  }
+` as typeof Autocomplete
+
+const StyledTextField = styled(TextField)`
+  legend {
+    letter-spacing: 0;
+  }
+` as typeof TextField
+
+export type LanguageChangePath = {
+  code: string
+  path: string | null
+  name: string
+}
+
+type LanguageListProps = {
+  languageChangePaths: LanguageChangePath[]
+  languageCode: string
+  close?: () => void
+  availableOnly?: boolean
+}
+
+export const filterLanguageChangePath = (
+  languageChangePath: LanguageChangePath,
+  query: string,
+  languageNamesInCurrentLanguage: Intl.DisplayNames,
+  languageNamesInFallbackLanguage: Intl.DisplayNames,
+): boolean => {
+  if (query === '') {
+    return true
+  }
+  const normalizedQuery = normalizeString(query)
+  return (
+    normalizeString(languageChangePath.name).includes(normalizedQuery) ||
+    normalizeString(languageNamesInCurrentLanguage.of(languageChangePath.code) || '').includes(normalizedQuery) ||
+    normalizeString(languageNamesInFallbackLanguage.of(languageChangePath.code) || '').includes(normalizedQuery)
+  )
+}
+
+const LanguageList = ({
+  languageChangePaths,
+  languageCode,
+  close,
+  availableOnly = false,
+}: LanguageListProps): ReactElement => {
+  const { t } = useTranslation('layout')
+  const { mobile } = useDimensions()
+  const [query, setQuery] = useState('')
+
+  const allOptions = useMemo(
+    () => languageChangePaths.filter(item => !availableOnly || item.path),
+    [languageChangePaths, availableOnly],
+  )
+
+  const currentLanguage = allOptions.find(item => item.code === languageCode)
+
+  const filteredLanguageChangePaths = useMemo(() => {
+    const languageNamesInCurrentLanguage = new Intl.DisplayNames([languageCode], { type: 'language' })
+    const languageNamesInFallbackLanguage = new Intl.DisplayNames([config.sourceLanguage], { type: 'language' })
+    return languageChangePaths.filter(item =>
+      filterLanguageChangePath(item, query, languageNamesInCurrentLanguage, languageNamesInFallbackLanguage),
+    )
+  }, [languageChangePaths, query, languageCode])
+
+  if (mobile) {
```

Comment:
❓ is there any reason why we don't use the auto complete component also for mobile?